msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/AbstractGuestAccountMsalBrokerUiTest.java

```diff
@@ -0,0 +1,83 @@
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.msal.automationapp.testpass.broker;
+
+import androidx.annotation.NonNull;
+
+import com.microsoft.identity.client.msal.automationapp.AbstractGuestAccountMsalUiTest;
+import com.microsoft.identity.client.msal.automationapp.BuildConfig;
+import com.microsoft.identity.client.ui.automation.annotations.SupportedBrokers;
+import com.microsoft.identity.client.ui.automation.broker.BrokerCompanyPortal;
+import com.microsoft.identity.client.ui.automation.broker.BrokerHost;
+import com.microsoft.identity.client.ui.automation.broker.BrokerMicrosoftAuthenticator;
+import com.microsoft.identity.client.ui.automation.broker.ITestBroker;
+import com.microsoft.identity.client.ui.automation.rules.RulesHelper;
+
+import org.junit.rules.RuleChain;
+
+import java.util.Arrays;
+import java.util.List;
+
+public abstract class AbstractGuestAccountMsalBrokerUiTest extends AbstractGuestAccountMsalUiTest {
+    protected ITestBroker mBroker = getBroker();
+
+    @Override
+    public RuleChain getPrimaryRules() {
+        return RulesHelper.getPrimaryRules(getBroker());
+    }
+
+    @NonNull
+    private ITestBroker getBroker() {
+        // only initialize once....so calling getBroker from anywhere returns the same instance
+        if (mBroker == null) {
+            mBroker = createBrokerFromFlavor();
+        }
+        return mBroker;
+    }
+
+    private ITestBroker createBrokerFromFlavor() {
+        switch (BuildConfig.SELECTED_BROKER) {
+            case BuildConfig.BrokerHost:
+                return new BrokerHost();
+            case BuildConfig.BrokerMicrosoftAuthenticator:
+                return new BrokerMicrosoftAuthenticator();
+            case BuildConfig.BrokerCompanyPortal:
+                return new BrokerCompanyPortal();
+            case BuildConfig.AutoBroker: {
+                final SupportedBrokers supportedBrokersAnnotation = getClass().getAnnotation(SupportedBrokers.class);
+                if (supportedBrokersAnnotation == null) {
+                    return new BrokerMicrosoftAuthenticator();
+                }
+                final List<Class<? extends ITestBroker>> supportedBrokerClasses =
+                        Arrays.asList(supportedBrokersAnnotation.brokers());
+                if (BuildConfig.FLAVOR_main.equals("dist") && supportedBrokerClasses.contains(BrokerCompanyPortal.class)) {
+                    return new BrokerCompanyPortal();
+                } else {
+                    return new BrokerMicrosoftAuthenticator();
+                }
+            }
+            default:
+                throw new UnsupportedOperationException("Unsupported broker :(");
+        }
+    }
```

Comment:
probably can move this method to a helper class (as I remember there is a similar method in another place)